projects/graph/src/App.js

```diff
@@ -28,25 +62,118 @@ class GraphView extends Component {
    * Render the canvas
    */
   updateCanvas() {
+    const graph = this.state.graph
     let canvas = this.refs.canvas;
     let ctx = canvas.getContext('2d');
     
     // Clear it
     ctx.fillStyle = 'white';
     ctx.fillRect(0, 0, canvasWidth, canvasHeight);
+    ctx.font="20px Georgia";
+    ctx.textAlign = "center";
+    
+    
 
-    // !!! IMPLEMENT ME
-    // compute connected components
-    // draw edges
-    // draw verts
-    // draw vert values (labels)
+    const connectedGraph = graph.getConnectedComponents()
+    ctx.lineWidth = 2;
+    for (let i = 0; i < connectedGraph.length; i++) {
+      const getRandomColor = () => {
```

Comment:
Random color function found on S/O. On review, I think I could have figured this out myself.